cmd/connect.go

```diff
@@ -73,30 +91,121 @@
 		}
 
-		// Note: The original `tsh` output is now manually printed above.
-		// The TUI will later capture and display `tsh` output for *new* connections made via the UI.
-
 		fmt.Printf("Current Kubernetes context set to: %s\n", teleportContextToUse)
-
-		// --- Print Initial Setup Info (can be displayed in TUI later) ---
 		fmt.Println("--------------------------")
-		fmt.Println("Setup complete. Initializing TUI...")
-		// Provider determination can be moved into the TUI model or passed if needed
-		// For now, keeping it simple and focusing on port-forward display
 
-		// Initialize the TUI model
-		tuiModel := tui.InitialModel(managementCluster, fullWorkloadClusterName, teleportContextToUse)
-		p := tea.NewProgram(tuiModel, tea.WithAltScreen()) // tea.WithOutput(os.Stderr) can be useful for debugging
+		if noTUI {
+			fmt.Println("Skipping TUI. Setting up port forwarding in the background...")
+			// Placeholder for non-TUI port forwarding logic
+			// This will involve calling a modified version of port forwarding setup
 
-		if _, err := p.Run(); err != nil {
-			fmt.Fprintf(os.Stderr, "Error running TUI: %v\n", err)
-			return err
-		}
+			// Get port forwarding configurations
+			configs := getPortForwardConfigs(managementCluster, fullWorkloadClusterName, teleportContextToUse)
+			if len(configs) == 0 {
+				fmt.Println("No port forwarding configurations found. Exiting.")
+				return nil
+			}
+
+			var wg sync.WaitGroup
+			stopChannels := make([]chan struct{}, 0)
+			allStopChan := make(chan struct{}) // Single channel to signal all goroutines
+
+			for _, pfConfig := range configs {
+				wg.Add(1)
+				// Use a local copy of pfConfig for the goroutine
+				config := pfConfig
+				go func() {
+					defer wg.Done()
+					fmt.Printf("Attempting to start port-forward for %s on %s to %s:%s (context: %s)...\n",
+						config.label, config.service, config.localPort, config.remotePort, config.kubeContext)
+
+					// Simple console logger for updates
+					sendUpdateFunc := func(status, outputLog string, isError, isReady bool) {
+						logPrefix := fmt.Sprintf("[%s] ", config.label)
+						if isError {
+							fmt.Printf("%sERROR: %s %s\n", logPrefix, status, outputLog)
+						} else if isReady {
+							fmt.Printf("%sREADY: %s %s\n", logPrefix, status, outputLog)
+						} else if outputLog != "" {
+							fmt.Printf("%sLOG: %s\n", logPrefix, outputLog)
+						} else if status != "" {
+							fmt.Printf("%sSTATUS: %s\n", logPrefix, status)
+						}
+					}
+
+					// StartPortForwardClientGo expects localPort:remotePort format
+					portSpec := fmt.Sprintf("%s:%s", config.localPort, config.remotePort)
+
+					// Start the port-forwarding
+					// Note: StartPortForwardClientGo returns (stopChan, initialStatus, initialError)
+					// We need to handle the initialStatus and initialError appropriately.
+					individualStopChan, initialStatus, initialErr := utils.StartPortForwardClientGo(
```

Comment:
The error handling here could be improved. Currently, if port forwarding setup fails, we just log the error and continue. Consider failing fast if critical services can't be port forwarded:

```go
if initialErr != nil {
    fmt.Fprintf(os.Stderr, "[%s] Failed to start port-forward: %v. Initial Status: %s\n", config.label, initialErr, initialStatus)
    // For critical services like Prometheus/Grafana, we may want to fail fast
    if config.isCritical {
        return fmt.Errorf("failed to start critical service %s: %w", config.label, initialErr)
    }
    return // Don't try to manage stopChan if setup failed
}
```